File: src/main/java/com/spinel/hr/base/domain/Customer.java
package com.spinel.hr.base.domain;

import javax.persistence.Entity;
import javax.persistence.Table;

import lombok.Data;
import lombok.EqualsAndHashCode;

@Entity
@Data
@EqualsAndHashCode(callSuper=false)
@Table(name="Customers")
public class Customer extends AbstractDomain {

	private String name;
	
	private Address address;
	
	private Location location;
	
	private Contact contact;
	
}

File: webapp/src/app/domain/customer/customer.ts
import { AbstractDomain } from '../common/abstract-domain';
import { Address } from '../common/address';
import { Contact } from '../common/contact';
export interface Customer extends AbstractDomain {

    name:string;
	
	address?:Address;

    location?:Location;

    contact?:Contact;
    
}
